src/coreclr/jit/morph.cpp

```diff
@@ -13550,6 +13545,26 @@ GenTree* Compiler::fgOptimizeEqualityComparisonWithConst(GenTreeOp* cmp)
             DEBUG_DESTROY_NODE(rshiftOp->gtGetOp2());
             DEBUG_DESTROY_NODE(rshiftOp);
         }
+
+        // Optimizes (X & 1) == 1 to (X & 1)
```

Comment:
Hmm, I think it'll be easier to reason about this if we move the (new) optimization just under the `op2->IsIntegralConst(0) || op2->IsIntegralConst(1)`, before the other ones. On success this returns early, on failure does nothing, so subsequent transformations don't need to take it into account.

Also, presumably we would need to check `op2Value == 1`.